addons/calendar/models/mail_message.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+from odoo.addons.calendar.models.calendar import get_real_ids
+
+
+class MailMessage(models.Model):
+    _inherit = "mail.message"
+
+    @api.model
+    def search(self, args, offset=0, limit=0, order=None, count=False):
+        '''
+        convert the search on real ids in the case it was asked on virtual ids, then call super()
+        '''
+        for index in range(len(args)):
+            if args[index][0] == "res_id":
+                if isinstance(args[index][2], basestring):
+                    args[index][2] = get_real_ids(args[index][2])
```

Comment:
Will it work, I think it should raise TypeError: 'tuple' object does not support item assignment